### PR TITLE
Rewrite with_module_fallback without format macro

### PR DESCRIPTION
**Pull request by killercup**
Also swaps the if/else branches to reduce indentation (and make the
possibly hotter non-unknown case ore obvious).

### CODE REVIEW

**Comment by killercup** [comment]
I think this was the last thing you talked about before we distracted you with other stuff :)

File: src/collapse-perf.rs [hunk 1]
```diff
@@ -325,23 +325,34 @@ impl PerfState {
...
-        } else {
-            format!("[{}]", rawfunc)
-        }
+    // try to use part of module name as function if unknown
+    let rawfunc = if module != "[unknown]" {
+        // use everything following last / of module as function name
+        &module[module.rfind('/').map(|i| i + 1).unwrap_or(0)..]
+    } else {
+        "unknown"
+    };
```
Comment (jonhoo) [inline src/collapse-perf.rs]: How about:
How about, above:
```
if module != "[unknown]" {
    // ...
} else if !include_addrs {
    return rawfunc.to_string();
} else {
    "unknown"
}
```
to save us the string construction below for the `rawfunc == module == "[unknown]"` and `!include_addrs` common-case? Might not matter in terms of performance though...

Comment (killercup) [reply src/collapse-perf.rs]: Good idea! I might want to write it like this, though :)

```rust
let rawfunc = match (module, include_addrs) {
    ("[unknown]", true) => "unknown",
    ("[unknown]", false) => {
        // no need to process this further
        return rawfunc.to_string();
    }
    (module, _) => {
        // use everything following last / of module as function name
        &module[module.rfind('/').map(|i| i + 1).unwrap_or(0)..]
    }
};
```

Comment (jonhoo) [reply src/collapse-perf.rs]: I like that! Either order is fine I think.

**Comment by jonhoo** [comment]
Thanks!

**Pull request merged**